woocommerce/woocommerce-admin
# Release process: Add build for Core

Partially addresses https://github.com/woocommerce/woocommerce-admin/issues/3176

In order to make wc-admin available for Core inclusion [Packagist](https://packagist.org/packages/woocommerce/woocommerce-admin) exposes the assets created in the source code zip found in releases. See [0.21.0](https://github.com/woocommerce/woocommerce-admin/releases/tag/v0.21.0) for example.

This PR adds an additional Core build and release to `npm run build:release` such that relevant builds with appropriate feature configs can be found in the source code zip.

## Test

Testing this involves creating two releases for plugin and core. One would do this step after `npm run pre-release` and `npm run changelog`, so we can assume version numbers are correct, docs have been updated, and you are on the correct release branch.

1. Make sure you have [hub](https://github.com/github/hub) via `brew install hub`
1. `npm run build:release`
3. `Please enter the version number to tag, for example, 1.0.0:` You can enter something like `0.23.0-test4`
5. Watch the script continue and eventually create the releases on Github with the zips attached
6. Find your [release on Github](https://github.com/woocommerce/woocommerce-admin/releases) and download the zips
8. Compare to make sure all the files and folder are the same in each
9. Check `includes/feature-config.php` in each. The core build has `onboarding` set to false, while the same file in the plugin build has the property set to true.

### Bonus testing

Try creating a custom build:

1. `npm run build:release -- --slug devdocs --features '{"devdocs":true}'`
2. Check the release on Github and inspect the zip's `includes/feature-config.php` file.
3. Devdocs should be turned on.
4. Ensure no Core build has been created.
